src/utils/ohdf/outputGenerator.ts

```diff
@@ -0,0 +1,293 @@
+ */
+export function extractResultSets(execJSONs: Record<string, ContextualizedEvaluation>, profileName: string): string[] {
+  logger.verbose('In extractResultSets')
+  return Object.entries(execJSONs).filter(([, execJSON]) => {
+    return execJSON.data.profiles[0].name === profileName
+  }).map(([filePath]) => {
+    return convertFullPathToFilename(filePath)
+  })
+}
+
+/**
+ * Generates a value for a specific cell in the Markdown table.
+ *
+ * The function works as follows:
+ * - It retrieves the data for the specified column from the item.
+ * - If the data exists and contains the specified key, it returns the value as a string.
+ * - If the data does not exist or does not contain the specified key, it returns '0'.
+ *
+ * @remarks
+ * This function is part of the `outputGenerator.ts` module.
+ *
+ * @param item - The PrintableSummary object containing the data for the table.
+ * @param column - The name of the column to retrieve the data for.
+ * @param key - The key to retrieve the value for from the column data.
+ * @returns A string representing the value for the cell.
+ */
+export function generateValue(item: PrintableSummary, column: string, key: string): string {
+  logger.debug('item:', item)
+  logger.debug('column:', column)
+  logger.debug('key:', key)
+
+  const columnData = item[column] as Record<string, number>
+
+  logger.debug('columnData:', columnData)
+
+  if (columnData && key in columnData) {
+    return columnData[key].toString()
+  }
+
+  return '0'
+}
+
+/**
+ * Generates a row for the Markdown table.
+ *
+ * The function works as follows:
+ * - It maps over the `COLUMN_ORDER` array, generating a value for each cell in the row using the `generateValue` function.
+ *
+ * @remarks
+ * This function is part of the `outputGenerator.ts` module.
+ *
+ * @param row - The name of the row to generate the data for.
+ * @param item - The PrintableSummary object containing the data for the table.
+ * @returns An array of strings, each representing a cell in the row.
+ */
+export function generateMarkdownTableRow(row: string, item: PrintableSummary): string[] {
+  return COLUMN_ORDER.map(column => generateValue(item, column, row))
+}
+
+/**
+ * Converts the provided data to a Markdown table.
+ *
+ * The table has a row for each value in `ROW_ORDER` and a column for each value in `COLUMN_ORDER`.
+ * The values in the table are extracted from the data object.
+ *
+ * @remarks
+ * This function is part of the `outputGenerator.ts` module.
+ *
+ * @param data - The data object or array of data objects containing the values for the table.
+ * @param titleTables - Boolean to either enable or disable adding titles to the produced markdown tables.
+ * @returns An array of strings, each representing a Markdown table.
+ */
+export function convertToMarkdown(data: DataOrArray, titleTables: boolean): string[] {
+  logger.verbose('In convertTomarkdown')
+  let tables: string[] = []
```

Comment:
could probably return the expression outright here instead of assigning to the 'tables' variable.